ACEHAL/fit.py

```diff
@@ -321,8 +321,7 @@ def do_fit(Psi, Y, B, E0s, solver, n_committee=8, basis_normalization=None, pot_
             solver.reset_threshold(threshold)
             coef_t = np.array(solver.coef_)
             residuals_t = Psi@coef_t-Y
-            K = np.sum(solver.var_c_ > (solver.var_c_min * solver.threshold))
-            BIC = n * np.log(np.mean(residuals_t ** 2)) + K * np.log(n)
+            BIC = n * np.log(np.mean(residuals_t ** 2)) + (solver.mask_ * np.log(n))
```

Comment:
This will fail if `fit` is called with sklearn's ARD, right? I'm probably going to add a test to check for that. I'll also refactor the code below to be used in both places.

Reply:
Also, you missed `sum()` call, right?

Reply:
It checks whether you're using the BayesianRegressionMax solver first as this one does allow you to adjust the threshold as a post processing step which is really quite useful as it keeps basis sizes optimal. 

This is mostly why I think we should drop the sklearn support actually.

Reply:
Yeah, I just noticed the test.  However, this diff changes the meaning of the `K` that's saved (now it's the full basis, not the ARD-selected subset) - I'm not sure which makes more sense.  Maybe both?  Also it's not a scalar because there's no sum of `mask_`, right?

Reply:
I don't like resetting the threshold without an argument to control it.  Can you take a break from changes there while I refactor that code?

Reply:
I think I have a better idea.  What do you think about moving the threshold optimization inside the solver?

Reply:
This looks great, I like how you took the range out of the code itself and for the user to specify.

I'll try it out now, thank you very much